File: python_backend/api_helpers/id_assign_/sub_ids.py
from pprint import pprint
from api_helpers.id_assign_.standard_ids import _standard_ids


def sub_ids(subs):
    """replace any of Bishop Knight or Queen in the game by assigning its id to the name of a custom piece.
       assign standard ids to custom piece names, resulting in a dict of form "standard_id":"custom_piece_name"
    """
    id_dict = {}
    for custom_piece_name in subs.keys():
        standard_piece_name = subs[custom_piece_name]
        standard_id = _standard_ids[standard_piece_name]
        id_dict[standard_id] = custom_piece_name
    return id_dict


if __name__ == "__main__":
    pprint(sub_ids({'Jester': 'Queen', 'Duke': 'Bishop'}))

File: node.js/apiHelpers/idAssign/subIds.js
const standardIds = require("./standardIds");


function subIds(subs) {
    /**replace any of Bishop Knight or the.includes(Queen) game by assigning its id to the name of a custom piece.
       assign standard ids to custom piece names, resulting in a dict of form "standardId":"customPieceName"
    */
    let idDict = {}
    for (var customPieceName of Object.keys(subs)) {
        let standardPieceName = subs[customPieceName]
        let standardId = standardIds[standardPieceName]
        idDict[standardId] = customPieceName
    }
    return idDict
}

module.exports = subIds;

if (require.main === module) {
    console.log(subIds({'Jester': 'Queen', 'Duke': 'Bishop'}))
}